### DISCUSSION TITLE
Two types of search in a table, one global search and one search within a filter due to tech limitation

### DISCUSSION THREAD

**Discussion by charleneyjh1** [twilio-labs/paste#3138] [Q&A] [answered]
Hello team, would like to get some help on a problem I encountered when designing for search in a table. 

I'm designing a Logs table for troubleshooting profile events. Based on customer and support feedback, there's usually two ids user usually use to search for an event, a message id and an external id. 

However, due to tech limitation, user has to identify an external id type first, before they can do the search by external id, and it has to be an exact match (no auto-population). Meanwhile, customer can directly search by a message id. 

To solve for this, I used a disclosure component and allow user to search within it.  
I'd like to get some feedback and guidance on the component/pattern usage here. Are there any products have the same issues?

Your help is much appreciated! 

<img width="1096" alt="Screenshot 2023-03-30 at 5 00 06 PM" src="https://user-images.githubusercontent.com/56697620/228983220-c29ec571-14a0-4aae-b14d-8bf36061a2d0.png">

**Comment by TheSisb** [comment]
Hi @charleneyjh1,

I see that you're signed up for office hours. Looking forward to discussing this in person. I'll update this discussion with our notes after the office hours meeting.

**Comment by TheSisb** [comment, marked as answer]
Small follow up from office hours.

We realized that the first row of inputs could alleviate the need for the second row potentially. If taking this approach, we could swap `External Id type` and `search` on the first row.  Some other notes included how we label the inputs and use placeholders, since they change based on external id / message type, and the possibility of adding more columns in the table so what is being searched is visibly reflected in the table.

Hope this helped!

**Comment by libertycarlin** [comment]
@charleneyjh1 

Here is the [loom recording](https://www.loom.com/share/afd4eb470a064a7d9dd889bb7d775ccb) of Office Hours and password is in the #Help-design-system channel.
